Source/JavaScriptCore/runtime/RegExpGlobalDataInlines.h

```diff
@@ -90,11 +90,15 @@ ALWAYS_INLINE void RegExpGlobalData::recordMatch(VM& vm, JSGlobalObject* owner,
     m_cachedResult.record(vm, owner, regExp, string, result);
 }
 
-inline void RegExpGlobalData::resetResultFromCache(JSGlobalObject* owner, RegExp* regExp, JSString* string, Vector<int>&& vector)
+inline MatchResult RegExpGlobalData::matchResult() const
+{
+    return m_cachedResult.result();
+}
+
+inline void RegExpGlobalData::resetResultFromCache(JSGlobalObject* owner, RegExp* regExp, JSString* string, MatchResult matchResult, Vector<int>&& vector)
 {
-    MatchResult result(vector[0], vector[1]);
```

Comment:
Can we turn this into a Debug ASSERT to help catch this issue if it still manifests?
```
ASSERT(matchResult.start == vector[0]);
ASSERT(matchResult.end == vector[1]);
```

Reply:
I don't think this condition is correct. ovector[0] and ovector[1] can be changed even if result is failing. So the above assertions are wrong (and that's the bug I think).

Reply:
When looking at the YarrJIT code, it is obviously wrong. I don't think we should add wrong ASSERT here. Let's just land this as is, and see the status. If the crash still appears, anyway, the above ASSERT is not related.